### PR TITLE
Make code more clear

### PR DESCRIPTION
**Pull request by ngaut**

### CODE REVIEW

File: tikv/mvcc.go [hunk 1]
```diff
@@ -58,20 +58,27 @@ func NewMVCCStore(db *badger.DB, dataDir string) *MVCCStore {
...
 	go store.writeLockWorker.run()
-	rbGCWorker := rollbackGCWorker{store: store}
-	go rbGCWorker.run()
+	go func() {
+		rbGCWorker := rollbackGCWorker{store: store}
+		rbGCWorker.run()
+	}()
+
+	// mark worker count
+	store.wg.Add(3)
```
Comment (coocood) [inline tikv/mvcc.go]: It's safer to add WaitGroup count before the goroutines run.

Comment (ngaut) [reply tikv/mvcc.go]: Why was that? we are still inside the current initialize function.

Comment (coocood) [reply tikv/mvcc.go]: In this case, it is safe, but in general coding style, we may run into wg count decrease to a negative value in some cases.

File: tikv/mvcc.go [hunk 2]
```diff
@@ -58,20 +58,27 @@ func NewMVCCStore(db *badger.DB, dataDir string) *MVCCStore {
...
 }
 
 func (store *MVCCStore) Close() error {
 	close(store.closeCh)
 	store.wg.Wait()
+
 	err := store.dumpLockStore()
 	if err != nil {
-		log.Error(err)
+		log.Fatal(err)
```
Comment (coocood) [inline tikv/mvcc.go]: We need to close the DB later.

Comment (ngaut) [reply tikv/mvcc.go]: dumpLockStore should never fail.

File: tikv/mvcc.go (hunk 1 above)
Comment (ngaut) [reply tikv/mvcc.go]: We should never use WaitGroup until we initialize it successful. I prefer to keep this clear.

**Pull request merged**